x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1659,6 +1659,68 @@ export class RulesClient {
     );
   }
 
+  public async unsnooze({ id }: { id: string }): Promise<void> {
+    return await retryIfConflicts(
+      this.logger,
+      `rulesClient.unsnooze('${id}')`,
+      async () => await this.unsnoozeWithOCC({ id })
+    );
+  }
+
+  private async unsnoozeWithOCC({ id }: { id: string }) {
+    const { attributes, version } = await this.unsecuredSavedObjectsClient.get<RawRule>(
+      'alert',
+      id
+    );
+
+    try {
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: attributes.alertTypeId,
+        consumer: attributes.consumer,
+        operation: WriteOperations.Unsnooze,
+        entity: AlertingAuthorizationEntity.Rule,
+      });
+
+      if (attributes.actions.length) {
+        await this.actionsAuthorization.ensureAuthorized('execute');
+      }
+    } catch (error) {
+      this.auditLogger?.log(
+        ruleAuditEvent({
+          action: RuleAuditAction.SNOOZE,
+          savedObject: { type: 'alert', id },
+          error,
+        })
+      );
+      throw error;
+    }
+
+    this.auditLogger?.log(
+      ruleAuditEvent({
+        action: RuleAuditAction.SNOOZE,
+        outcome: 'unknown',
+        savedObject: { type: 'alert', id },
+      })
+    );
```

Comment:
I missed that you added a new "Snooze" RuleAuditAction a few days ago.
This is a separate action, can you please do the following?

1. Add a new action in `x-pack/plugins/alerting/server/rules_client/audit_events.ts` for unsnooze
2. Change these two audit events to use the new unsnooze action
3. Change the docs in `docs/user/security/audit-logging.asciidoc` to describe _both_ new audit actions

----

Also: I'm not sure if the intent here is to snooze the _rule itself_ or to snooze the alert that has been generated _by_ the rule.
If it's the latter, the audit action for snooze/unsnooze should reflect that (it should be `SNOOZE_ALERT`/`rule_alert_snooze`...)

I'm happy to re-review once you make changes to the `audit_events.ts` file.